connection/journey.test.js

```diff
@@ -1,9 +1,59 @@
-import { AuthUserPasswordCredentials, AuthAccessTokenCredentials } from './auth.js';
+import {
+  AuthUserPasswordCredentials, 
+  AuthAccessTokenCredentials, 
+  AuthClientCredentials
+} from './auth.js';
 import Connection from "./index.js";
 
 const weaviate = require("../index");
 
 describe("connection", () => {
+  it("makes an Azure logged-in request with client credentials", async() => {
+    if (process.env.AZURE_CLIENT_SECRET == undefined || process.env.AZURE_CLIENT_SECRET == "") {
+      console.warn("Skipping because `AZURE_CLIENT_SECRET` is not set");
+      return;
+    }
+
+    const client = weaviate.client({
+      scheme: "http",
+      host: "localhost:8084",
+      authClientSecret: new AuthClientCredentials({
+        clientSecret: process.env.AZURE_CLIENT_SECRET
+      })
+    })
+
+    return client.misc
+      .metaGetter()
+      .do()
+      .then((res) => {
+        expect(res.version).toBeDefined();;
+      })
+      .catch((e) => fail("it should not have errord: " + e));
+  })
+
+  it("makes an Okta logged-in request with client credentials", async() => {
+    if (process.env.OKTA_CLIENT_SECRET == undefined || process.env.OKTA_CLIENT_SECRET == "") {
+      console.warn("Skipping because `OKTA_CLIENT_SECRET` is not set");
+      return;
+    }
+
+    const client = weaviate.client({
+      scheme: "http",
+      host: "localhost:8081",
+      authClientSecret: new AuthClientCredentials({
+        clientSecret: process.env.OKTA_CLIENT_SECRET
+      })
+    })
+
+    return client.misc
+      .metaGetter()
+      .do()
+      .then((res) => {
+        expect(res.version).toBeDefined();;
+      })
+      .catch((e) => fail("it should not have errord: " + e));
+  })
```

Comment:
The two tests are basically the same, so parametrizing would be great (see [python example](https://github.com/weaviate/weaviate-python-client/blob/7a512ea1177fa839b79335fc62daa6ecac53c995/integration/test_authentication.py#L49))
(not sure if JS supports that)

Reply:
Sure, maybe worth refactoring later